mundane-dsv/src/test/scala/com/ambiata/mundane/dsv/DelimitedValuesParserSpec.scala

```diff
@@ -0,0 +1,73 @@
+package com.ambiata.mundane.dsv
+
+import com.ambiata.disorder._
+import Arbitraries._
+import org.scalacheck._, Gen._, Arbitrary._
+import org.specs2._
+import org.specs2.matcher.DisjunctionMatchers
+
+class DelimitedValuesParserSpec extends Specification with ScalaCheck with DisjunctionMatchers { def is = s2"""
+
+ A parser for delimited fields can deal with
+   comma-separated values $csv
+   quoted fields          $quoted
+   malformed lines        $malformed
+
+                                                                          """
+
+  def csv = prop { csvLine: CsvLine =>
+    DelimitedValuesParser.csv.parse(csvLine.value) must be_\/-(csvLine.expected)
+  }
+
+  def quoted = prop { csvLine: CsvQuotedLine =>
+    DelimitedValuesParser.csv.parse(csvLine.value) must be_\/-(csvLine.expected)
+  }
+
+  def malformed = prop { csvLine: CsvMalformedLine =>
+    DelimitedValuesParser.csv.parse(csvLine.value) must be_-\/
+  }
+
+  case class CsvLine(value: String, expected: List[String])
+
+  implicit def ArbitraryCsvLine: Arbitrary[CsvLine] =
+    Arbitrary(Gen.sized(n => listOfN(n, genRegularField).map(fs => CsvLine(fs.mkString(","), fs))))
+
+  case class CsvQuotedLine(value: String, expected: List[String])
+
+  implicit def ArbitraryCsvQuotedLine: Arbitrary[CsvQuotedLine] =
+    Arbitrary(Gen.sized(n => listOfN(n, genQuotedField)
+      .map(fs => 
+        CsvQuotedLine(
+          if (fs.isEmpty) "" 
+          else fs.mkString("\"", "\",\"", "\""), 
+          fs.map(_.replace("\"\"", "\"")))))) // expected double-quotes to be removed
+
+  case class CsvMalformedLine(value: String)
+
+  implicit def ArbitraryCsvMalformedLine: Arbitrary[CsvMalformedLine] =
+    Arbitrary(Gen.sized(n => listOfN(n+1, genMalformedField) // generate at least one malformed field
+      .map(fs =>
+      CsvMalformedLine(
+        if (fs.isEmpty) ""
+        else fs.mkString("\"", "\",\"", "\""))))) // expected double-quotes to be removed
+
+
+  def genRegularField: Gen[String] =
+    arbitrary[Ident].map(_.value)
+
+  def genQuotedField: Gen[String] =
+   for {
+     q <- Gen.oneOf("\"\"", ",", "") // q might be a double quote a delimiter or empty
+     a <- genRegularField
+     b <- genRegularField
+   } yield s"$a$q$b"
+
+  def genMalformedField: Gen[String] =
+    for {
+      q <- Gen.oneOf("\"") // a non-escaped quote will fail the parse
+      a <- genRegularField
+      b <- genRegularField
+    } yield s"$a$q$b"
+
+
+}
```

Comment:
The level of confidence required for this is fairly high, I would probably want to see some larger scale tests, perhaps a code generator and a known corpus of a bunch of documents that we can trivially add to if we come up against unusual edge cases and differences in behaviour between implementations.